mm2src/coins/utxo/slp.rs

```diff
@@ -459,40 +431,49 @@ impl SlpToken {
         match slp_tx.transaction {
             SlpTransaction::Send { token_id, amounts } => {
                 if token_id != self.token_id() {
-                    return MmError::err(ValidateHtlcError::InvalidSlpDetails);
+                    return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                        "Invalid tx token_id, Expected: {}, found: {}",
+                        token_id,
+                        self.token_id()
+                    )));
                 }
 
                 if amounts.is_empty() {
-                    return MmError::err(ValidateHtlcError::InvalidSlpDetails);
+                    return MmError::err(ValidatePaymentError::WrongPaymentTx(
+                        "Input amount can't be empty".to_string(),
+                    ));
                 }
 
                 if amounts[0] != slp_satoshis {
-                    return MmError::err(ValidateHtlcError::InvalidSlpDetails);
+                    return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
+                        "Invalid input amount. Expected: {}, found: {}",
+                        slp_satoshis, amounts[0]
+                    )));
                 }
             },
-            _ => return MmError::err(ValidateHtlcError::InvalidSlpDetails),
+            _ => {
+                return MmError::err(ValidatePaymentError::InvalidPaymentTxData(
```

Comment:
Not sure also if this should be `WrongPaymentTx` or `InvalidPaymentTxData`, I am leaning towards `WrongPaymentTx` since this means that `SlpTransaction` enum is not the `Send` variant.

Reply:
I think you're right, it's better use `WrongPaymentTx` as the transaction **is not** an invalid